File: MiniDOMWebCtrls/DMD/CQSPD/CQSPDDashInfo.cs
using DMD.XML;

namespace minidom.Forms
{
    public class CQSPDDashInfo 
        : IDMDXMLSerializable
    {
        public int Count;
        public decimal SommaMontanteLordo;
        public decimal SommaProvvigioneTotale;
        public decimal SommaSpread;
        public decimal SommaUpFront;
        public decimal SommaRunning;
        public CCollection<CQSPDPDBInfo> Pratiche;

        public CQSPDDashInfo()
        {
            DMDObject.IncreaseCounter(this);
            Count = 0;
            SommaMontanteLordo = 0m;
            SommaProvvigioneTotale = 0m;
            SommaSpread = 0m;
            SommaUpFront = 0m;
            SommaRunning = 0m;
            Pratiche = new CCollection<CQSPDPDBInfo>();
        }

        protected void XMLSerialize(XMLWriter writer)
        {
            writer.WriteAttribute("Count", Count);
            writer.WriteAttribute("SommaMontanteLordo", SommaMontanteLordo);
            writer.WriteAttribute("SommaProvvigioneTotale", SommaProvvigioneTotale);
            writer.WriteAttribute("SommaSpread", SommaSpread);
            writer.WriteAttribute("SommaUpFront", SommaUpFront);
            writer.WriteAttribute("SommaRunning", SommaRunning);
            writer.WriteTag("Pratiche", Pratiche);
        }

        protected void SetFieldInternal(string fieldName, object fieldValue)
        {
            switch (fieldName ?? "")
            {
                case "Pratiche":
                    {
                        Pratiche = (CCollection<CQSPDPDBInfo>)fieldValue;
                        break;
                    }

                case "Count":
                    {
                        Count = (int)DMD.XML.Utils.Serializer.DeserializeInteger(fieldValue);
                        break;
                    }

                case "SommaMontanteLordo":
                    {
                        SommaMontanteLordo = (decimal)DMD.XML.Utils.Serializer.DeserializeDouble(DMD.Strings.CStr(fieldValue));
                        break;
                    }

                case "SommaProvvigioneTotale":
                    {
                        SommaProvvigioneTotale = (decimal)DMD.XML.Utils.Serializer.DeserializeDouble(DMD.Strings.CStr(fieldValue));
                        break;
                    }

                case "SommaRunning":
                    {
                        SommaRunning = (decimal)DMD.XML.Utils.Serializer.DeserializeDouble(DMD.Strings.CStr(fieldValue));
                        break;
                    }

                case "SommaSpread":
                    {
                        SommaSpread = (decimal)DMD.XML.Utils.Serializer.DeserializeDouble(DMD.Strings.CStr(fieldValue));
                        break;
                    }

                case "SommaUpFront":
                    {
                        SommaUpFront = (decimal)DMD.XML.Utils.Serializer.DeserializeDouble(DMD.Strings.CStr(fieldValue));
                        break;
                    }
            }
        }

        ~CQSPDDashInfo()
        {
            DMDObject.DecreaseCounter(this);
        }

        void IDMDXMLSerializable.XMLSerialize(XMLWriter writer)
        {
            this.XMLSerialize(writer);
        }

        void IDMDXMLSerializable.SetFieldInternal(string fieldName, object fieldValue)
        {
            this.SetFieldInternal(fieldName, fieldValue);
        }

    }
}

File: MiniDOMWebCtrls/DMD/CQSPD/CQSPDDashInfo.vb
Imports minidom
Imports minidom.Forms
Imports minidom.Databases
Imports minidom.WebSite

Imports minidom.Finanziaria
Imports minidom.Sistema
Imports minidom.Anagrafica

Namespace Forms



    Public Class CQSPDDashInfo
        Implements XML.IDMDXMLSerializable

        Public Count As Integer
        Public SommaMontanteLordo As Decimal
        Public SommaProvvigioneTotale As Decimal
        Public SommaSpread As Decimal
        Public SommaUpFront As Decimal
        Public SommaRunning As Decimal

        Public Pratiche As CCollection(Of CQSPDPDBInfo)

        Public Sub New()
            DMDObject.IncreaseCounter(Me)
            Me.Count = 0
            Me.SommaMontanteLordo = 0
            Me.SommaProvvigioneTotale = 0
            Me.SommaSpread = 0
            Me.SommaUpFront = 0
            Me.SommaRunning = 0
            Me.Pratiche = New CCollection(Of CQSPDPDBInfo)
        End Sub

        Protected Sub XMLSerialize(ByVal writer As minidom.XML.XMLWriter) Implements XML.IDMDXMLSerializable.XMLSerialize
            writer.WriteAttribute("Count", Me.Count)
            writer.WriteAttribute("SommaMontanteLordo", Me.SommaMontanteLordo)
            writer.WriteAttribute("SommaProvvigioneTotale", Me.SommaProvvigioneTotale)
            writer.WriteAttribute("SommaSpread", Me.SommaSpread)
            writer.WriteAttribute("SommaUpFront", Me.SommaUpFront)
            writer.WriteAttribute("SommaRunning", Me.SommaRunning)
            writer.WriteTag("Pratiche", Me.Pratiche)
        End Sub

        Protected Sub SetFieldInternal(ByVal fieldName As String, ByVal fieldValue As Object) Implements XML.IDMDXMLSerializable.SetFieldInternal
            Select Case (fieldName)
                Case "Pratiche" : Me.Pratiche = fieldValue
                Case "Count" : Me.Count = XML.Utils.Serializer.DeserializeInteger(fieldValue)
                Case "SommaMontanteLordo" : Me.SommaMontanteLordo = XML.Utils.Serializer.DeserializeDouble(fieldValue)
                Case "SommaProvvigioneTotale" : Me.SommaProvvigioneTotale = XML.Utils.Serializer.DeserializeDouble(fieldValue)
                Case "SommaRunning" : Me.SommaRunning = XML.Utils.Serializer.DeserializeDouble(fieldValue)
                Case "SommaSpread" : Me.SommaSpread = XML.Utils.Serializer.DeserializeDouble(fieldValue)
                Case "SommaUpFront" : Me.SommaUpFront = XML.Utils.Serializer.DeserializeDouble(fieldValue)
            End Select
        End Sub

        Protected Overrides Sub Finalize()
            MyBase.Finalize()
            DMDObject.DecreaseCounter(Me)
        End Sub
    End Class


End Namespace